scipy/stats/_stats_py.py

```diff
@@ -6466,6 +6466,9 @@ def _t_confidence_interval(df, t, confidence_level, alternative, dtype=None, xp=
     dtype = t.dtype if dtype is None else dtype
     xp = array_namespace(t) if xp is None else xp
 
+    # stdtrit not dispatched yet; use NumPy
+    df, t = np.asarray(df), np.asarray(t)
```

Comment:
The warning is probably more general... I think it's some implicit conversion from`Tensor` to array. Looks like that was going on here - `df` and `t` were `xp`-arrays, but they are passed right into `stdtrit` (which isn't dispatched) below. Solution (I hope) is to explicitly convert to NumPy.